pym/_emerge/actions.py

```diff
@@ -1644,8 +1644,30 @@ def action_info(settings, trees, myopts, myfiles):
 
 	for repo in repos:
 		last_sync = portage.grabfile(os.path.join(repo.location, "metadata", "timestamp.chk"))
+		head_commit = None
 		if last_sync:
 			append("Timestamp of repository %s: %s" % (repo.name, last_sync[0]))
+		last_sync = portage.grabfile(os.path.join(repo.location, "metadata", "timestamp.commit"))
+		if last_sync:
+			try:
+				head_commit = last_sync[0].split()[0]
+			except IndexError:
+				pass
+		elif repo.sync_type == "git":
```

Comment:
SyncManager calls `portage.sync.module_controller.get_class(repo.sync_type)`, and the class constructor takes zero arguments. The new method will have to call `SyncBase._kwargs` to initialize self.repo.